dvc/command/pipeline.py

```diff
@@ -61,7 +61,7 @@ def _build_graph(self, target, commands, outs):
                     for out in stage.outs:
                         edges.append((str(out), str(dep)))
         else:
-            for from_stage, to_stage in networkx.dfs_edges(G, target_stage):
+            for from_stage, to_stage in networkx.edge_dfs(G, target_stage):
```

Comment:
This is not a complaint or anything like that, but just wanted to say that this is actually a really good case where a PR/patch description would help understand the difference :) Just like a short note about "we are currently using dfs_edges, but it is not what we need because ... , so we need to use edge_dfs". Since you dove into researching this, your summary would have a great value :slightly_smiling_face:

Reply:
Thats a valid point, Ill do that, it will probably be the best to include that in original issue, right?

Reply:
@pared, could this be related with this: https://github.com/iterative/dvc/issues/3353?

Reply:
@skshetry I don't think so. This issue was purely about `dvc pipelines`, and in #3353 throw happens upon `reproduce`. I would need to dive deeper into #3353 to say more.